martin/src/cog/image.rs

```diff
@@ -155,10 +155,10 @@ fn ensure_pixels_valid(
                     //   So the visual result is the same, but the component values are different.
 
                     let value = data[(idx_chunk + component_idx) as usize];
-                    if let Some(v) = nodata {
-                        if value == v {
-                            continue 'outer;
-                        }
+                    if let Some(v) = nodata
+                        && value == v
```

Comment:
🤔 I think this should be the following instead

```suggestion
                    if nodata == Some(value)
```